### PR TITLE
Rtl upgrade

### PR DESCRIPTION
**Pull request by AlexanderKaran**
- Upgraded all the tests from enzyme to react testing library to make upgrading to React 18 easier
- Updated every test to RTL
- Moved towards more testing patterns supported by RTL

### CODE REVIEW

File: src/resources/controllers/resource-subscriber/test.tsx [hunk 1]
```diff
@@ -98,26 +96,24 @@ describe('<ResourceSubscriber />', () => {
       },
     });
 
-    const Component = () => (
+    render(
       <ResourceSubscriber resource={mockResource}>
         {({ data, loading }) => {
           if (loading) {
-            return <div id="loading" />;
+            return <div data-testid="loading" />;
```
Comment (SpanishPear) [inline src/resources/controllers/resource-subscriber/test.tsx:103]: nit: probably better to do 
```suggestion
            return <div>loading</div>;
```
and query by text 

not a blocker though

File: src/ui/route-component/test.tsx [hunk 2]
```diff
@@ -33,18 +34,19 @@ const routes = [
...
 
-    const component = wrapper.find(MockComponent);
+    // Check if the mock component is rendered
+    const component = screen.getByText('My component');
+    expect(component).toBeInTheDocument();
 
-    expect(component).toHaveLength(1);
-
-    expect(component.prop('location')).toEqual(MockLocation);
+    // Optionally, check for location properties if they are passed as props
```
Comment (SpanishPear) [inline src/ui/route-component/test.tsx:49]: nit: This comment seems to describe something that isn't there?

File: package.json [hunk 3]
```diff
@@ -90,8 +90,8 @@
     "@codeshift/test-utils": "^0.3.0",
     "@codeshift/utils": "^0.2.0",
     "@testing-library/jest-dom": "^5.7.0",
-    "@testing-library/react": "^10.0.4",
-    "@types/enzyme": "^3.10.5",
+    "@testing-library/react": "^10.4.9",
```
Comment (SpanishPear) [inline package.json:93]: nit: are we able to upgrade to `14.0.0` (or whatever newest version is without anything breaking? Wonder if we can get best of both worlds ;)

**Comment by SpanishPear** [comment]
Looks good! I'll ping someone else to approve but ✅ from me (some nits, but they are nice to haves, not blocking)

**Review by SpanishPear — DISMISSED** [review]

**Comment by AlexanderKaran** [comment]
> Looks good! I'll ping someone else to approve but ✅ from me (some nits, but they are nice to haves, not blocking)

Thanks for the review will fix up the comments you mentioned today

File: src/controllers/router-store/test.tsx [hunk 4]
```diff
@@ -548,12 +548,12 @@ describe('RouterStore', () => {
         const RouteName = () => <>{useRouteName()}</>;
 
         const route = {
-          component: () => null,
+          component: () => <p>home</p>,
```
Comment (pancaspe87) [inline src/controllers/router-store/test.tsx:551]: Any particular reason why this was changed?

File: src/controllers/router/test.tsx [hunk 5]
```diff
@@ -78,24 +80,24 @@ describe('<Router />', () => {
...
       listen.mockReturnValue(unlisten1);
 
-      const wrapper = mount(
+      const { rerender } = render(
         <RemountingParent>
           <Router history={history} routes={routes} plugins={[]} />
         </RemountingParent>
       );
 
-      // first listener is created on mount
```
Comment (pancaspe87) [inline src/controllers/router/test.tsx:87]: Can we preserve these comments?

File: src/ui/link/test.tsx [hunk 6]
```diff
@@ -29,173 +30,154 @@ const defaultProps = {
...
   });
 
   it('should render a <Link />', () => {
-    const wrapper = mountInRouter('my link');
-    const anchor = wrapper.find('a');
-
-    expect(anchor.prop('href')).toEqual(defaultProps.href);
-  });
-
-  it('should support the `to` prop', () => {
```
Comment (pancaspe87) [inline src/ui/link/test.tsx:74]: Why did we remove this?

File: src/ui/route-component/test.tsx (hunk 2 above)
Comment (AlexanderKaran) [reply src/ui/route-component/test.tsx]: Removing comment as the test has changed

File: package.json (hunk 3 above)
Comment (AlexanderKaran) [reply package.json:93]: Tried to install the latest version; however it throws an error due to React16 being in the dependent tree

```
npm install --save-dev @testing-library/react@14.1.2
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: react-resource-router@0.25.0
npm ERR! Found: react@16.14.0
npm ERR! node_modules/react
npm ERR!   dev react@"^16.10.2" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^18.0.0" from @testing-library/react@14.1.2
npm ERR! node_modules/@testing-library/react
npm ERR!   dev @testing-library/react@"14.1.2" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! /Users/akaran/.npm/_logs/2023-11-22T02_25_00_700Z-eresolve-report.txt
```

File: src/controllers/router-store/test.tsx [hunk 7]
```diff
@@ -567,7 +567,7 @@ describe('RouterStore', () => {
           expect.objectContaining({ route }),
           undefined
         );
-        expect(wrapper.html()).toEqual('home');
```
Comment (bobrovnik) [inline src/controllers/router-store/test.tsx:570]: I wonder why was html==='home'? I thought `component: () => null,` should render `null`

File: src/controllers/router-store/test.tsx (hunk 4 above)
Comment (AlexanderKaran) [reply src/controllers/router-store/test.tsx:551]: The approach to testing is very different in RTL vs enzyme. The biggest being "test from the perspective of a user". I had to change the test to work by checking for what was rendered.

File: src/ui/link/test.tsx (hunk 6 above)
Comment (AlexanderKaran) [reply src/ui/link/test.tsx:74]: Sorry for my mistake. So many tests in this file and they need a big re-work from enzyme to RTL. Added back in

File: src/controllers/router-store/test.tsx (hunk 7 above)
Comment (AlexanderKaran) [reply src/controllers/router-store/test.tsx:570]: Not sure about the old tests but in the new one I am rendering something with the text home in it to make sure things are correctly rendered.

**Review by MonicaOlejniczak — APPROVED** [review]

**Review by SpanishPear — APPROVED** [review]
nice!

@MonicaOlejniczak are you able to merge? I don't have perms 😮‍💨

File: package.json (hunk 3 above)
Comment (SpanishPear) [reply package.json:93]: ah makes sense, 14 is dependent on 18

**Review by tomgasson — APPROVED** [review]

**Pull request merged**